lib/offline/storage.js

```diff
@@ -1334,10 +1334,19 @@ shaka.offline.Storage = class {
       if (!toDownload.has(pendingSegmentRefId)) {
         const estimateId = downloader.addDownloadEstimate(
             estimator.getSegmentEstimate(stream.id, segment));
+        let groupId = stream.id;
+        const numberOfParallelSegmentsByStream =
+            config.offline.numberOfParallelSegmentsByStream;
+        if (numberOfParallelSegmentsByStream > 1) {
+          // Note: 1000 is a random number to not coincide with other groups
+          // of other streams
+          groupId += 1000 + Math.floor(Math.random() *
+              numberOfParallelSegmentsByStream);
```

Comment:
If the goal is to have, say, 5 parallel tracks of audio and 5 of video, this won't do that.  For example:

video stream, id 1 => 1001 - 1005
audio stream, id 2 => 1002 - 1006
audio stream, id 3 => 1003 - 1007
text stream, id 4 => 1004 - 1008

So there will be mixing between groups.

However, why not mix between groups?  Why not assign each segment to a random group in the range of the config variable?

```js
const groupId = Math.floor(
    Math.random() * numberOfParallelSegmentsByStream);
```

Now that I'm looking at the problem this way, it may be better to change the config from `numberOfParallelSegmentsByStream` to `numberOfParallelDownloads`.  It's not "by stream" any more, nor did it ever really need to be.

The consistency of your results would also improve if:

1. groupIds were assigned in a round-robin manner instead of purely random
2. the default were set to something like 4 or 5 instead of 1

I would then call this a "feat" instead of a "fix", since the default would be different than the old behavior.  But feat vs fix here is somewhat subjective.  I wouldn't insist.

Reply:
Done!